website_seo_redirection/models/website_seo_redirection.py

```diff
@@ -42,6 +54,21 @@ def _check_destination(self):
         self._url_format_check("destination")
 
     @api.multi
+    @api.constrains("origin", "destination")
+    def _check_not_recursive(self):
+        """Avoid infinite loops."""
+        for redirection in self:
+            origins = {redirection.origin}
+            while redirection:
+                redirection = self.search([
+                    ("origin", "=", redirection.destination),
+                ])
+                if redirection.origin in origins:
```

Comment:
[It's impossible.](https://github.com/OCA/website/pull/246/files/6b9a8e5494aa80e8897c5cc8b22e737ca93d5552#diff-d011fe32656e50bc9cf8dc81716ac66aR19)